src/components/Header/index.ts

```diff
@@ -0,0 +1,46 @@
+import logoImage from "../../../templates/logo.png";
+import { $ } from "../../utils/selector";
+
+export class Header {
+  #$target;
+
+  constructor(
+    $target: Element,
+    onSubmitSearchKeyword: (searchKeyword: string) => void,
+    onClickLogoImage: () => void
+  ) {
+    this.#$target = $target;
+
+    this.render();
+
+    $(".search-box")?.addEventListener("submit", (event: Event) => {
+      event.preventDefault();
+
+      const $searchInput = $(".search-input");
+
+      if ($searchInput === null) return;
+
+      if ($searchInput instanceof HTMLInputElement) {
+        const inputValue = $searchInput.value;
+
+        if (inputValue === "") return alert("검색값을 입력해주세요.");
```

Comment:
```suggestion
        if (inputValue === "") {
          alert("검색값을 입력해주세요.");
          return;
        }
```

이렇게 작성하는 방법도 있는데, 뭐가 더 낫다고 생각하세요? (궁금)

Reply:
한줄이라 간결하게 보이려고 썼지만 다른 if문과 같이 볼 때는 가독성이 떨어지는 것 같습니다..ㅎㅎ

Reply:
alert는 값을 반환하지 않는 사이드이펙트함수입니다. 
반환값이 없는 함수를 return 하게 되면, 다른 개발자들은 이걸 보고 다른 값을 반환해야 하는데 실수로 alert를 쓴건가? 라는 생각을 하게 됩니다. 

따라서, 이런건 따로 써주는게 더 의도를 명확하게 드러낸다고 생각해요